pyEpiabm/pyEpiabm/sweep/intervention_sweep.py

```diff
@@ -72,9 +86,20 @@ def __call__(self, time):
             num_cases = sum(map(lambda cell: cell.number_infectious(),
                             self._population.cells))
             if intervention.is_active(time, num_cases):
-                intervention(time)
                 if self.intervention_active_status[intervention] is False:
+                    # Update with the single intervention
+                    # (with respect to index)
+                    if hasattr(intervention, 'occurrence_index'):
+                        intervention_type = \
+                            list(self.intervention_dict.keys())[list(
+                                self.intervention_dict.values()).index(
+                                type(intervention))]
+                        Parameters.instance().intervention_params[
```

Comment:
Similarly I think these lines are very long and not particularly readable - I'm thinking of ways to tidy it up but for now it would be good to comment what this does if possible